main.tf

```diff
@@ -1,14 +1,31 @@
-module "vpc" {
-  source = "./modules/vpc"
+locals {
+  dd_tags = {
+    Datadog                 = "true"
+    DatadogAgentlessScanner = "true"
+  }
+}
+
+resource "aws_secretsmanager_secret" "api_key" {
+  count = var.api_key != null ? 1 : 0
+  name  = "datadog-agentless-scanner-api-key"
+  tags  = merge(var.tags, local.dd_tags)
+}
 
+resource "aws_secretsmanager_secret_version" "api_key_version" {
+  count         = var.api_key != null ? 1 : 0
+  secret_id     = aws_secretsmanager_secret.api_key.id
+  secret_string = var.api_key
+}
+
```

Comment:
Let's think about multiregion set-up

Reply:
Indeed. Putting this in draft until I figure out the way to go